fix(ScyllaYaml): replace enable_tablets with tablets_mode_for_new_keyspaces

cezarmoise opened the pull request:
The configuration option was changed in https://github.com/scylladb/scylladb/pull/22273
Causing all SCT runs after it to run without tablets.
e.g. https://argus.scylladb.com/tests/scylla-cluster-tests/a50f6974-da50-4963-b7ea-bbfeac27bdca
2025.1 not affected **yet**, backport was yesterday.

### Testing
<!-- Add links to Argus/Jenkins of test test done with this PR -->
<!-- This would help the reviewer to cross check what was tested, and and review the results as needed -->
- [x] https://argus.scylladb.com/tests/scylla-cluster-tests/f04c9b13-6038-40b1-bd47-835a4c75a84f
  `2025.2.0~dev-20250403.fe8187e5940e`, before the change, keyspace1 has tablets enabled
- [x] https://argus.scylladb.com/tests/scylla-cluster-tests/b5f009d5-41b3-4130-8d1b-0436a62584a7
  `2025.2.0~dev-20250407.8d2a41db8231`, after the change, keyspace1 does not have tablets enabled
- [x] https://argus.scylladb.com/tests/scylla-cluster-tests/a89cfa9e-bce3-49fd-8929-210006d8b31f
  `2025.2.0~dev-20250407.8d2a41db8231`, after the change, with this fix, keyspace1 has tablets enabled

### PR pre-checks (self review)
<!--- PR should be created as Draft, when CI finished and relevant checkboxes selected, add reviewers and then click on "Ready for review" button.-->
<!--- Put an `x` in all the boxes that apply or create PR and then click on all relevant checkboxes: -->
- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

cezarmoise commented:
https://github.com/scylladb/scylla-cluster-tests/commit/f00c4982cbb114255b87790ad4c06a8645b9feed updated relevant unit test

soyacz commented:
@cezarmoise please search codebase for `enable_tablets` and adjust accordingly.
Still I wonder if we shouldn't keep `enable_tablets` setting to be able to run SCT with versions before master/2025.1.x.

roydahan commented:
I agree, we may still need to support the older param name.

Also, we need to backport to perf branches.

@juliayakovlev FYI

juliayakovlev commented:
> I agree, we may still need to support the older param name.
> 
> Also, we need to backport to perf branches.
> 
> @juliayakovlev FYI


Yes, we need to backport it to `perf-v16` and `perf-v15`

cezarmoise commented:
> What is the default value of `tablets_mode_for_new_keyspaces`?

`enabled`

soyacz commented:
@cezarmoise we still need `enable_tablets` setting to be in SCT - see upgrade tests start off from prior versions that support this (so we must support both).

cezarmoise commented:
https://github.com/scylladb/scylla-cluster-tests/commit/1b69c40c3fbc2d231c829f035fb48004a60203c7 replaced with new option name everywhere.

roydahan commented:
We need to support both types of parameters, old one and new one.
In cases we need to bisect or compare an issue with an older build, we need SCT to do it for the user.

@cezarmoise and @scylladb/qa-maintainers please give this one high priority as it breaks SCT jobs quite bad.

cezarmoise commented:
https://github.com/scylladb/scylla-cluster-tests/commit/dd70ef207e44a3823d323e50d1001b515728cb4e
> If `tablets_mode_for_new_keyspaces` is unset, then `enable_tablets: true` is equivalent to `tablets_mode_for_new_keyspaces: enabled` and `enable_tablets: false` is equivalent to `tablets_mode_for_new_keyspaces: disabled`
If `tablets_mode_for_new_keyspaces` is set, then `enable_tablets` is ignored.

Added both options everywhere.
Newer Scylla will use the new one as it takes precedence, older Scylla will ignore it and use `enable_tablets`.

cezarmoise commented on test-cases/longevity/longevity-aws-custom-d2-workload1-3dcs.yaml:
```diff
@@ -26,6 +26,7 @@ append_scylla_yaml:
   # See: https://github.com/scylladb/scylladb/pull/21207
   enable_small_table_optimization_for_rbno: true
   enable_tablets: false
+  tablets_mode_for_new_keyspaces: "disabled"
```
fixed in https://github.com/scylladb/scylla-cluster-tests/commit/973e24cdb23df3c0ec3854a7440ba7cb3861cbbd
It does make a difference
' is raw string, " can have stuff like \n in it.

soyacz reviewed (dismissed):
LGTM

aleksbykov commented on configurations/force-gossip-topology-changes.yaml:
```diff
@@ -1,3 +1,4 @@
 append_scylla_yaml:
   force_gossip_topology_changes: true
   enable_tablets: false
+  tablets_mode_for_new_keyspaces: 'disabled'
```
Did you test this config?
Scylla could failed to start if unsupported parameter in scylla.yaml or passed as parameter

cezarmoise commented:
Some tests (ran a simple 3h longevity until some data is written, checked Grafana, then aborted)
- https://argus.scylladb.com/tests/scylla-cluster-tests/2fbc051b-b49d-476d-a134-c98ff615253b
2025.2.0\~dev-20250407.8d2a41db8231 
tablets are enabled
- https://argus.scylladb.com/tests/scylla-cluster-tests/a017363d-512d-48a7-aecf-4bbb3674226e
2025.2.0\~dev-20250407.8d2a41db8231
added `configurations/tablets_disabled.yaml`
tablets are disabled
- https://argus.scylladb.com/tests/scylla-cluster-tests/6d734388-92d0-4134-85d0-0aea98dac1d1
2025.2.0\~dev-20250403.fe8187e5940e 
tablets are enabled
- https://argus.scylladb.com/tests/scylla-cluster-tests/a2215a49-ed25-468b-b1c6-237fd891392a
2025.2.0\~dev-20250403.fe8187e5940e 
added `configurations/tablets_disabled.yaml`
tablets are disabled


Scylla versions were chosen to be with and without the commit mentioned in the PR description

cezarmoise commented:
ping @scylladb/qa-maintainers

vponomaryov commented on configurations/tablets-initial-32.yaml:
```diff
@@ -1,3 +1,4 @@
 append_scylla_yaml:
-  enable_tablets: true
+  enable_tablets: false
```
The value for the enable_tablets option is inverted, is it intentional?

cezarmoise commented:
> @cezarmoise did you check why some checks are failing?

docker.errors.BuildError: The command '/bin/bash -c curl -L https://download.docker.com/linux/centos/docker-ce.repo -o /etc/yum.repos.d/docker-ce.repo' returned a non-zero code: 23

23 means write error.

But I don't see how it is related to my changes.
